libcontainer/specconv/spec_linux.go

```diff
@@ -405,6 +405,9 @@ func CreateCgroupConfig(opts *CreateOpts) (*configs.Cgroup, error) {
 		if r.CPU != nil {
 			if r.CPU.Shares != nil {
 				c.Resources.CpuShares = *r.CPU.Shares
+
+				//CpuWeight is used for cgroupv2 and should be converted
+				c.Resources.CpuWeight = (1 + ((*r.CPU.Shares-2)*9999)/262142)
```

Comment:
We should do this with a function so people know it's converting runtime-spec items to cgroupv2 value.

And comments like "convert from [2-262144] to [1-10000]" would be helpful, more explaination such as value 262144 comes from Linux kernel definition "#define MAX_SHARES               (1UL << 18)" will also help.

Reply:
These have been collected by @giuseppe for crun [here](https://github.com/giuseppe/crun/blob/master/crun.1.md#cgroup-v2) already. I'm thinking about how to go about adding these to the runtime-spec as per the discussion in opencontainers/runtime-spec#1002.

Should the functions you mention be part of the spec already?